jwst/lib/dispaxis.py

```diff
@@ -0,0 +1,152 @@
+import logging
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+def get_dispersion_direction(exposure_type, grating="ANY", filter="ANY",
+                             pupil="ANY"):
+    """Get the dispersion direction.
+
+    Parameters
+    ----------
+    exposure_type : str
+        The exposure type.
+
+    grating : str
+        The name of the optic in the grating wheel.
+
+    filter : str
+        The name of the optic in the filter wheel.
+
+    pupil : str
+        The name of the optic in the pupil wheel.
+
+    Returns
+    -------
+    int : The dispersion direction
+
+-       0  the dispersion direction is not meaningful or not defined
+-       1  the dispersion direction is horizontal ("sky" coordinates)
+-       2  the dispersion direction is vertical ("sky" coordinates)
+    """
+
+    exposure_type = exposure_type.upper()
+    grating = grating.upper()
+    filter = filter.upper()
+    pupil = pupil.upper()
+
+    # The keys of the `by_exp_type` dictionary are values of exposure type.
+    # If the dispersion direction is uniquely determined by the exposure
+    # type, the associated value of by_exp_type[exposure_type] will be an
+    # integer, the dispersion direction.  If one or more parameters other
+    # than the exposure type are needed, the value of
+    # by_exp_type[exposure_type] will be a tuple of several strings; use
+    # that tuple as the key for dictionary `second_pass`, and the value of
+    # that will be the dispersion direction.
+    by_exp_type = {
+        # FGS science
+        "FGS_DARK": 0,
+        "FGS_FOCUS": 0,
+        "FGS_IMAGE": 0,
+        "FGS_INTFLAT": 0,
+        "FGS_SKYFLAT": 0,
+        # FGS guide star
+        "FGS_ACQ1": 0,
+        "FGS_ACQ2": 0,
+        "FGS_FINEGUIDE": 0,
+        "FGS_ID-IMAGE": 0,
+        "FGS_ID-STACK": 0,
+        "FGS_TRACK": 0,
+        # MIRI
+        "MIR_4QPM": 0,
+        "MIR_CORONCAL": 0,
+        "MIR_DARKALL": 0,
+        "MIR_DARKIMG": 0,
+        "MIR_DARKMRS": 0,
+        "MIR_FLATALL": 0,
+        "MIR_FLATIMAGE": 0,
+        "MIR_FLATIMAGE-EXT": 0,
+        "MIR_FLATMRS": 2,
```

Comment:
Should this be 0? It's not going through assign_wcs.

Reply:
Given that it's not going through assign_wcs probably implies that we don't need to cover this mode (it'll never go through any level-2b processing), but I think the setting is still accurate (i.e. the data in the flat will be dispersed along axis 2).